<?xml version="1.0"?>

<!-- Launch the UR10. -->
<launch>
  <arg name="use_joy_stick_relay" default="false" doc="use joy stick for emergency stop"/>

  <!-- this is ultimately remapped to /ur10/cmd_vel -->
  <remap from="joint_group_vel_controller/command" to="cmd_vel" unless="$(arg use_joy_stick_relay)"/>
  <remap from="joint_group_vel_controller/command" to="ur10_velocity_controller/cmd_vel" if="$(arg use_joy_stick_relay)"/>

  <!-- Launch UR10 driver. -->
  <!-- namespace is required to avoid conflict with Ridgeback control topics/services -->
  <include file="$(find ur_robot_driver)/launch/ur10_bringup.launch" ns="ur10">
    <arg name="robot_ip" default="192.168.131.40"/>
    <arg name="kinematics_config" default="$(find mobile_manipulation_central)/config/ur10/kinematics_parameters.yaml"/>
    <arg name="robot_description_file" value="$(find mobile_manipulation_central)/launch/thing_description.launch" />
    <arg name="controller_config_file" value="$(find mobile_manipulation_central)/config/ur10/controllers.yaml" />
    <arg name="controllers" value="joint_state_controller joint_group_vel_controller" />

    <!-- TODO would like to get rid of this -->
    <arg name="stopped_controllers" value="scaled_vel_joint_traj_controller" />
  </include>
</launch>
